File: api/src/main/java/gg/mineral/api/nbt/ByteArrayTag.java
package gg.mineral.api.nbt;

import lombok.val;

/**
 * The {@code TAG_Byte_Array} tag.
 */
public final class ByteArrayTag extends Tag<byte[]> {

    /**
     * The value.
     */
    private final byte[] value;

    /**
     * Creates the tag.
     *
     * @param value The value.
     */
    public ByteArrayTag(byte... value) {
        super(TagType.BYTE_ARRAY);
        this.value = value;
    }

    @Override
    public byte[] getValue() {
        return value;
    }

    @Override
    protected void valueToString(StringBuilder hex) {
        for (byte b : value) {
            val hexDigits = Integer.toHexString(b & 0xff);
            int len = 1;
            if (hexDigits.length() == len)
                hex.append("0");

            hex.append(hexDigits).append(" ");
        }
    }

}

File: api/src/main/kotlin/gg/mineral/api/nbt/ByteArrayTag.kt
package gg.mineral.api.nbt

/**
 * The `TAG_Byte_Array` tag.
 */
class ByteArrayTag
/**
 * Creates the tag.
 *
 * @param value The value.
 */(
    /**
     * The value.
     */
    override vararg val value: Byte
) : Tag<ByteArray>(TagType.BYTE_ARRAY) {

    override fun valueToString(builder: StringBuilder) {
        for (b in value) {
            val hexDigits = Integer.toHexString(b.toInt() and 0xff)
            val len = 1
            if (hexDigits.length == len) builder.append("0")

            builder.append(hexDigits).append(" ")
        }
    }
}
